### PR TITLE
Harmony 1378 online documentation

### PR DESCRIPTION
**Pull request by indiejames**
## Jira Issue ID
HARMONY-1378

## Description
Adds a documentation page

## Local Test Steps
1. Check out this branch
2. run harmony locally
3. go to the landing page
4. click on the 'Documentation' link
5. Read the documentation
6. Verify that all the harmony links are to localhost (should use deployment environment)
7. Verify that all EDL links are to UAT (will be prod EDL in Harmony prod)
8. Bear in mind that this is not intended to be complete just yet. we can add things as we need them

## PR Acceptance Checklist
* [x] Acceptance criteria met
* [x] Tests added/updated (if needed) and passing
* [x] Documentation updated (if needed)

### CODE REVIEW

File: app/markdown/endpoints.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## Summary of Available Endpoints
+
+All of the public endpoints for Harmony users other than the OCG Coverages and WMS APIs are listed in the following table. The Coverages and WMS APIs are described in the next section.
+
+| route                  | description                                                                       |
+|------------------------|-----------------------------------------------------------------------------------|
+| /                      | The Harmony landing page                                                          |
+| /cloud-access          | Generates JSON with temporary credentials for accessing processed data in S3      |
+| /cloud-access.sh       | Generates shell scripts that can be run to access processed data in S3             |
```
Comment (chris-durbin) [inline app/markdown/endpoints.md:9]: no clue if the extra space before the end of the row matters

**Review by chris-durbin — APPROVED** [review]
Looks good - verified that all the links I could find worked. One other comment that I couldn't figure out where it came from in the code is that I think the "Documentation" displayed at the top of the page takes up too much room - can we make that smaller?

File: app/frontends/docs.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+/**
+ * Express.js handler that returns the Harmony documentation page content.
+ *
+ * @param req - The request sent by the client
+ * @param res - The response to send to the client
+ */
+export default async function docsPage(req: HarmonyRequest, res: Response): Promise<void> {
+  const root = getRequestRoot(req);
+  docsHtml = null;
```
[the remark this answers is not in the document]
Comment (indiejames) [reply app/frontends/docs.ts:83]: Good catch. This was caching I turned off while developing and forgot to reenable.

File: app/markdown/overview.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+## Overview
+Harmony provides access to services that can transform data from NASA's [Earth Observing Systems Data and Information System (EOSDIS)](https://www.earthdata.nasa.gov/eosdis/) Distributed Active Archive Centers (DAAC).
+Transformations can be requested using one of two [Open Geospatial Consortium (OCG)](https://www.ogc.org/) inspired APIs.
```
[the remark this answers is not in the document]
Comment (indiejames) [reply app/markdown/overview.md:3]: Fixed

File: app/frontends/interpolation-markdown-it-plugin.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import MarkDownIt from 'markdown-it';
```
[the remark this answers is not in the document]
Comment (indiejames) [reply app/frontends/interpolation-markdown-it-plugin.ts:1]: Done

File: app/frontends/docs.ts [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+  if (!docsHtml) {
+    let { tableCount, exampleCount } = await getTableAndExampleCounts();
```
[the remark this answers is not in the document]
Comment (indiejames) [reply app/frontends/docs.ts:85]: Done

**Comment by indiejames** [comment]
> Looks good - verified that all the links I could find worked. One other comment that I couldn't figure out where it came from in the code is that I think the "Documentation" displayed at the top of the page takes up too much room - can we make that smaller?

I reduced the font size and height of the title div.

**Pull request merged**